docs/adr/0031-process-evolution.md

```diff
@@ -0,0 +1,171 @@
+that release process by defining what properties we’d like to see in a process
+and then determining the best way to get there.
+
+### Requirements of a release process include
+
+- Needs to be robust. This means resilient to human error (or indifferent to
+  certain decisions). We can rollback changes and there isn’t a failure state
+  where production is half-pushed.
+- Needs to allow stakeholders to do their work in a reasonable time and manner
+- Bias towards fewer steps a human needs to do
+
+### Stakeholders and their expectations:
+
+#### SRE
+
+- Expects reliable software. SRE is currently on pager-duty.
+- Expects good communication about significant changes. We currently document
+  configuration and database changes in a deployment document.
+
+#### Engineers
+
+- Expects a reliable and expedient process
+- Expects an easy way to know the current status of any environment
+
+#### QA
+
+- Expects good communication about changes
+
+#### L10n
+
+- Expects strings accessible in the L10n tooling. Currently this is
+  pontoon.mozilla.org.
+- Expects minimal changes to strings once available
+- Expects strings available with enough lead time to localize them before a
+  major feature ships. We cannot ship major features which aren’t localized.
+
+#### Relying parties
+
+- Expects a stable platform and high availability
+
+## Considerations
+
+### Do we need to push all 8 servers all the time?
+
+If we allow the servers to run different versions of code we need to be
+mindful of:
+
+- Careful API versioning to ensure the servers can continue to communicate with each other
+- Careful database updates if more than one server is accessing the database directly
+- Differences between staging and production when testing
+- An easy way to know what version each server is running
+
+In light of the above it’s probably best to continue to keep all servers pushed
+to their latest versions. _That doesn’t necessarily mean the same version_, but
+if a newer version exists it should go out.
+
+### Do we need tags?
+
+When we are ready to deploy our code we tag it (eg. v1.100.0) as an easy means
+of identifying the code we are pushing. Tags are just aliases in git so our
+usage is purely for ease of use. That said, it is significantly easier to
+remember a tag name than a hash which makes it easier to identify what is live,
+what was live at an arbitrary time in the past, and if we need to roll back,
+easier to get the right code live. Since they offer benefits and no real
+drawback we should continue to use tags.
+
+### Who uses the change logs and what are their expectations?
+
+We’ve maintained change logs in each package for many years and they are
+updated with each release. We release often enough that not all of the
+packages change between each release and we regularly update the changelogs
+just to say nothing has changed. This causes considerable churn in our
+deployment pipeline for no reason.
```

Comment:
Thanks for the feedback.  By deployment pipeline churn I mean we're deploying new packages and servers when the only change is an update to the changelog that says "no changes"

Reply:
Ahh, that makes sense. For some reason I thought we didn't redeploy those packages, I could have sworn I've checked `/__version__` before and had to be reminded that only reflected content-server's versioning when I'd done a patch release and then wasn't sure if the release had made it to staging since I expected the version number to be higher. 🤔

Reply:
That does happen but it's when SRE is doing things by hand